sotu.md

```diff
@@ -11,138 +11,158 @@ The topics are roughly sorted from greatest strengths to greatest weaknesses. Ea
+  - [Others](http://caml.inria.fr/cgi-bin/hump.en.cgi?sort=0&browse=88) - Some other compilers implemented in OCaml, quite a few dead now.
 
-- **Rating**: ...
-- **Notable libraries**: ...
-- **Educational resources**: ...
+- **Parser and Lexer Generators**:
+  - [Menhir](http://gallium.inria.fr/~fpottier/menhir) – Menhir is a refined (LA)LR(1) parser generator for OCaml.
+  - [dypgen](http://dypgen.free.fr/) – Self-extensible GLR parser generator
+  - [ocamllex](http://caml.inria.fr/pub/docs/manual-ocaml-4.01/lexyacc.html) – lex (and yacc) implementation for OCaml.
+  - [Opal](https://github.com/pyrocat101/opal) – Self-contained monadic parser combinators for OCaml.
+  - [DeCaP](http://lama.univ-savoie.fr/decap/) – Parser generator (aiming at efficiency)
 
+- **Articles**:
+  - [Kaleidoscope: Implementing a Language with LLVM in Objective Caml¶](http://llvm.org/docs/tutorial/OCamlLangImpl1.html)
+  - [Getting started with OCaml bindings for LLVM](http://nopaniers.calepin.co/getting-started-with-ocaml-bindings-for-llvm.html)
 
-## Compilers and Compiler Tools
+## Formal Verification
 
-- **Rating**: ...
-- **Notable libraries**: ...
-- **Educational resources**: ...
+- **Rating**: Best in class
 
+- **Notable projects**:
 
-## Concurrency
+  - [Coq](http://coq.inria.fr/) – Coq is a formal proof management system. It provides a formal language to write mathematical definitions, executable algorithms and theorems together with an environment for semi-interactive development of machine-checked proofs.
+  - [Why3](http://why3.lri.fr/) – Why3 is a platform for deductive program verification. It provides a rich language for specification and programming, called WhyML, and relies on external theorem provers, both automated and interactive, to discharge verification conditions.
+  - [Frama-C](http://frama-c.com/what_is.html): static analysis toolbox for the C language
+  - [TLA+](https://tla.msr-inria.inria.fr/tlaps/content/Download/Source.html): modeling and verification of concurrent and distributed systems
+  - [Astrée](http://www.astree.ens.fr/): abstract-analysis based program analyzer
+  - [Alt-Ergo](http://alt-ergo.lri.fr/) – Alt-Ergo is an open-source SMT solver dedicated to the proof of mathematical formulas generated in the context of program verification.
 
-- **Rating**: ...
-- **Notable libraries**: ...
 - **Educational resources**: ...
 
+## Scientific Computing
 
-## Databases
-
-- **Rating**: ...
+- **Rating**: Immature
 - **Notable libraries**: ...
 - **Educational resources**: ...
 
+## Web Development
 
-## Developer Tools
-
-- **Rating**: ...
+- **Rating**: Immature
 - **Notable libraries**: ...
 - **Educational resources**: ...
 
+## Mobile Applications
 
-## Formal Software Verification
-
-- **Rating**: ...
+- **Rating**: Immature
 - **Notable libraries**: ...
 - **Educational resources**: ...
 
+## System Programming
 
-## Graphics
-
-- **Rating**: ...
+- **Rating**: Immature
```

Comment:
Agreed. I would even say OCaml is one of the finest system programming languages, since it keeps a prefect balance between security and flexibility of the low level primitives.